listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -622,6 +675,64 @@ def get_listen_events(
     return events
 
 
+def get_listen_events_new(
+    users: List[Dict],
+    min_ts: int,
+    max_ts: int,
+    limit: int
+) -> List[APITimelineEvent]:
+    """ Gets all listen events in the feed.
+    """
+    # To avoid timeouts while fetching listen events, we want to make
+    # sure that both min_ts and max_ts are defined. if only one of those
+    # is set, calculate the other from it using a default window length.
+    # if neither is set, use current time as max_ts and subtract window
+    # length to get min_ts.
+    if min_ts and max_ts:
+        temp = datetime.utcfromtimestamp(min_ts)
+        max_ts = datetime.utcfromtimestamp(max_ts)
+        if max_ts - temp < DEFAULT_LISTEN_EVENT_WINDOW_NEW:
+            min_ts = temp 
+        else:
+            # If the given interval for search is greater than :DEFAULT_LISTEN_EVENT_WINDOW_NEW:,
+            # then we must limit the search interval to :DEFAULT_LISTEN_EVENT_WINDOW_NEW:.
+            min_ts = max_ts - DEFAULT_LISTEN_EVENT_WINDOW_NEW
+    else:
```

Comment:
You can remove this nesting and directly use elif here.

Reply:
Done